src/Database/Connection.php

```diff
@@ -841,9 +842,9 @@ public function setLogger(?QueryLogger $logger)
     /**
      * Gets the logger object
      *
-     * @return \Cake\Database\Log\QueryLogger logger instance
+     * @return \Psr\Log\LoggerInterface logger instance
      */
-    public function getLogger(): QueryLogger
+    public function getLogger(): LoggerInterface
     {
         if ($this->_logger === null) {
             $this->_logger = new QueryLogger();
```

Comment:
This is now a potentially breaking change for us eres of the database package, since QueryLogger extends something that may not be there

Reply:
If we remove `cakephp/log` as hard dependency of `cakephp/database` then it would be up to the user to either add `cakephp/log` as dependency for their app or use package of their choice and use `setLogger()` to explicitly set a logger instance after creating connection instance.

`getLogger()` can do a check whether a logger is already set and if not first ensure `BaseLog` class is available before trying to create `QueryLogger` instance else throw exception.

Reply:
Or if we don't mind having extra files pulled in even if user doesn't want to user our logger, then we keep `cakephp/log` as dependency thereby achieving backward compatibility as nothing changes for end user.

Reply:
We could also do a class check on `Cake\Log\Log` and fallback to `error_log()` which dumps to the SAPI error output, which is stderr in CLI and webserver error logs in apache/fpm.

I'm also ok with adding cake/log as an `suggest` dependency. We could throw an error here if `Cake\Log\Log` doesn't exist.